tests/sentry/ingest/test_transaction_rule_validator.py

```diff
@@ -22,3 +22,13 @@ def test_non_all_star_rules_valid():
 
     assert RuleValidator(ReplacementRule("a/*/b/*/c/**")).is_valid()
     assert RuleValidator(ReplacementRule("/a/*/b/*/c/**")).is_valid()
+
+
+def test_schema_all_stars_invalid():
+    assert not RuleValidator(ReplacementRule("http://*/*/*/**")).is_valid()
+    assert not RuleValidator(ReplacementRule("https://*/*/*/**")).is_valid()
+    assert RuleValidator(ReplacementRule("http://a/*/*/**")).is_valid()
+    assert RuleValidator(ReplacementRule("http://example.com/*/*/**")).is_valid()
+    assert RuleValidator(
+        ReplacementRule("http://user:password@www.example.com:80/*/*/*/**")
```

Comment:
Should we add some more cases here, like `ftp://`, `file:///`, no schema, ...

Reply:
Added a few more tests with `ftp` and `file` in [6bfe7f9](https://github.com/getsentry/sentry/pull/62076/commits/6bfe7f962cfba76084d78b6412f980b66b5adb24). I think the test above covers the no-schema cases well enough.